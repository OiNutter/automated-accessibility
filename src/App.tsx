import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import Button from './components/Button'
import Logo from './components/Logo'

function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      <header role="banner">
        <a href="https://vitejs.dev" title="Vite" target="_blank">
          <Logo src={viteLogo} alt="Vite logo" />
        </a>
        <a href="https://react.dev" title="React" target="_blank">
          <Logo src={reactLogo} alt="React logo" />
        </a>
      </header>
      <main>
      <h1>Vite + React</h1>
      <div className="card">
        <Button onClick={() => setCount((count) => count + 1)}>
          count is {count}
        </Button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      </main>
      <footer role="contentinfo">
        <p className="read-the-docs">
          Click on the Vite and React logos to learn more
        </p>
      </footer>
    </>
  )
}

export default App
