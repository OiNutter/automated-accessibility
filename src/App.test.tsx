import { render } from "@testing-library/react"
import {axe} from "vitest-axe"
import App from "./App"
import { virtual } from "@guidepup/virtual-screen-reader";

describe("App", () => {
  
  it("should render an accessible page", async () => {
    const { container } = render(<App/>)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it("should be screen reader friendly", async (t) => {
    const {container} = render(<App/>)
    t.onTestFinished(async () => await virtual.stop())
    await virtual.start({container})
    await virtual.next() // Vite Link
    expect(await virtual.lastSpokenPhrase()).toEqual("link, Vite logo, Vite")

    await virtual.next() // Vite Logo
    await virtual.next() // End of Vite Link
    await virtual.next() // React Link

    expect(await virtual.lastSpokenPhrase()).toEqual("link, React logo, React")
    
    await virtual.next() // React Logo
    await virtual.next() // End of React Link
    await virtual.next() // End of banner
    await virtual.next() // Main content
    await virtual.next() // Heading
    expect(await virtual.lastSpokenPhrase()).toEqual("heading, Vite + React, level 1")

    await virtual.next() // Button
    expect(await virtual.lastSpokenPhrase()).toEqual("button, count is 0")
    await virtual.press("Enter")
    expect(await virtual.lastSpokenPhrase()).toEqual("button, count is 1")

    expect(await virtual.spokenPhraseLog()).toMatchSnapshot()

  })

})