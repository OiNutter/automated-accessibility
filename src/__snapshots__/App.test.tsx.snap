// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should be screen reader friendly 1`] = `
[
  "banner",
  "link, Vite logo, Vite",
  "img, Vite logo",
  "end of link, Vite logo, Vite",
  "link, React logo, React",
  "img, React logo",
  "end of link, React logo, React",
  "end of banner",
  "main",
  "heading, Vite + React, level 1",
  "button, count is 0",
  "button, count is 1",
]
`;
